viewer/packages/sector-loader/src/GltfSectorRepository.ts

```diff
@@ -47,67 +49,72 @@ export class GltfSectorRepository implements SectorRepository {
-          });
-          break;
-        case RevealGeometryCollectionType.TriangleMesh:
-          this.createMesh(group, parsedGeometry.geometryBuffer, materials.triangleMesh);
-          break;
-        default:
-          assertNever(type);
-      }
-    });
-
-    return {
-      levelOfDetail: sector.levelOfDetail,
-      group: group,
-      instancedMeshes: [],
-      metadata: metadata,
-      modelIdentifier: sector.modelIdentifier,
-      geometryBatchingQueue: geometryBatchingQueue
-    };
+      const group = new AutoDisposeGroup();
+
+      const parsedSectorGeometry = this._gltfSectorParser.parseSector(sectorByteBuffer);
+
+      const materials = this._materialManager.getModelMaterials(sector.modelIdentifier);
+
+      const geometryBatchingQueue: ParsedGeometry[] = [];
+
+      parsedSectorGeometry.forEach(parsedGeometry => {
+        const type = parsedGeometry.type as RevealGeometryCollectionType;
+
+        const filteredGeometryBuffer = filterGeometryOutsideClipBox(
+          parsedGeometry.geometryBuffer,
+          type,
+          sector.geometryClipBox ?? undefined
+        );
+
+        if (!filteredGeometryBuffer) return;
+
+        switch (type) {
+          case RevealGeometryCollectionType.BoxCollection:
+          case RevealGeometryCollectionType.CircleCollection:
+          case RevealGeometryCollectionType.ConeCollection:
+          case RevealGeometryCollectionType.EccentricConeCollection:
+          case RevealGeometryCollectionType.EllipsoidSegmentCollection:
+          case RevealGeometryCollectionType.GeneralCylinderCollection:
+          case RevealGeometryCollectionType.GeneralRingCollection:
+          case RevealGeometryCollectionType.QuadCollection:
+          case RevealGeometryCollectionType.TorusSegmentCollection:
+          case RevealGeometryCollectionType.TrapeziumCollection:
+          case RevealGeometryCollectionType.NutCollection:
+            geometryBatchingQueue.push({ type, geometryBuffer: filteredGeometryBuffer, instanceId: type.toString() });
+            break;
+          case RevealGeometryCollectionType.InstanceMesh:
+            geometryBatchingQueue.push({
+              type,
+              geometryBuffer: filteredGeometryBuffer,
+              instanceId: parsedGeometry.instanceId!
+            });
+            break;
+          case RevealGeometryCollectionType.TriangleMesh:
+            this.createMesh(group, parsedGeometry.geometryBuffer, materials.triangleMesh);
+            break;
+          default:
+            assertNever(type);
+        }
+      });
+
+      return {
+        levelOfDetail: sector.levelOfDetail,
+        group: group,
+        instancedMeshes: [],
+        metadata: metadata,
+        modelIdentifier: sector.modelIdentifier,
+        geometryBatchingQueue: geometryBatchingQueue
+      };
+    } catch (error) {
+      MetricsLogger.trackError(error as Error, { moduleName: 'GltfSectorRepository', methodName: 'loadSector' });
```

Comment:
I noticed while working with the "not resuming loading after reconnect"-problem that the error passed to MetricsLogger from the V8SectorRepository always(?) was `undefined`. Have you checked whether the error you get here is more meaningful?

Reply:
On closer inspection, the error seems to be the same as the one received in V8SectorRepository, which is just `undefined`. I'm fixing this in my own PR, so no action is required here.